src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -63,6 +69,118 @@ private static String getFileName(String uri) {
             }
         }
 
+        private String getConfigurationUrl() {
+            YarnConfiguration conf = new YarnConfiguration();
+            String httpPolicy = conf.get(YARN_HTTP_POLICY);
+            if (httpPolicy != null && httpPolicy.equals(YARN_HTTP_POLICY_HTTPS_ONLY)) {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+                }
+                return "https://" + address + "/conf";
+            } else {
+                String address = conf.get(YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+                if (address == null || address.isEmpty()) {
+                    address = conf.get(YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+                }
+                return "http://" + address + "/conf";
+            }
+        }
+
+        private Protos.CommandInfo getCommandInfo() {
+            MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
+            CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
+            if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
+                /*
+                 Overall this is messier than I'd like due to Mesos 1790. We can't let mesos untar the distribution,
+                 since it will change the permissions.  Instead we simply download the tarball and execute tar -xvpf.
+                 We also pull the config from the resource manager and put them in the conf dir.  This is also why we need
+                 frameworkSuperUser
+                */
+
+                //Too much work checking all posibilities and many are likely to fail somewhere later anyway
+                //Keep in mind this is a temp fix anyway
```

Comment:
Again, leave a TODO so we know to remove this "temp fix" when it is no longer needed. (How will we know?)